### PR TITLE
Change Build Framework & add application package

### PR DESCRIPTION
**Pull request by jijoongmoon**
- From now on, we are going to use meson as an build framework instead
of cmake.
- Add nntrainer-applications packages

**Self evaluation:**
1. Build test:	 [X]Passed [ ]Failed [ ]Skipped
2. Run test:	 [X]Passed [ ]Failed [ ]Skipped

Signed-off-by: jijoong.moon <jijoong.moon@samsung.com>

### CODE REVIEW

File: packaging/nntrainer.spec [hunk 1]
```diff
@@ -9,7 +10,7 @@ License:        Apache-2.0
 Source0:        nntrainer-%{version}.tar.gz
 Source1001:     nntrainer.manifest
 
-BuildRequires:  cmake >= 2.8.3
+BuildRequires:	meson >= 0.50.0
 BuildRequires:  openblas-devel
 BuildRequires:  iniparser-devel
```
Comment (dongju-chae) [inline packaging/nntrainer.spec]: I think tab is required here too.

Comment (jijoongmoon) [reply packaging/nntrainer.spec]: done. Thanks

File: meson.build [hunk 2]
```diff
@@ -0,0 +1,106 @@
...
+if get_option('enable-blas')
+  add_project_arguments('-DUSE_BLAS=1', language:['c','cpp'])
+  if build_platform == 'tizen'
+    blas_dep = dependency('openblas')
+  else
+    blas_dep = dependency('blas-openblas')
+  endif
+endif
+
+libm_dep = cc.find_library('m') # cmath library
```
Comment (wooksong) [inline meson.build]:
```suggestion
libm_dep = cxx.find_library('m') # cmath library
```

File: meson.build [hunk 3]
```diff
@@ -0,0 +1,106 @@
...
+libdl_dep = cc.find_library('dl') # DL library
```
Comment (wooksong) [inline meson.build]:
```suggestion
libdl_dep = cxx.find_library('dl') # DL library
```

It seems that there are no c source files.

File: meson.build (hunk 2 above)
Comment (jijoongmoon) [reply meson.build]: Fixed!

File: meson.build (hunk 3 above)
Comment (jijoongmoon) [reply meson.build]: Fixed!

**Pull request merged**